nni/algorithms/compression/pytorch/pruning/iterative_pruner.py

```diff
@@ -120,14 +119,16 @@ class AGPPruner(IterativePruner):
 
     def __init__(self, model, config_list, optimizer, trainer, criterion,
                  num_iterations=10, epochs_per_iteration=1, pruning_algorithm='level'):
-        super().__init__(model, config_list, optimizer=optimizer, trainer=trainer, criterion=criterion,
+        self.freq = epochs_per_iteration
+        self.end_epoch = epochs_per_iteration * num_iterations
+
+        super().__init__(model, config_list, optimizer=optimizer, pruning_algorithm=pruning_algorithm, trainer=trainer, criterion=criterion,
                          num_iterations=num_iterations, epochs_per_iteration=epochs_per_iteration)
         assert isinstance(optimizer, torch.optim.Optimizer), "AGP pruner is an iterative pruner, please pass optimizer of the model to it"
-        self.masker = MASKER_DICT[pruning_algorithm](model, self)
+
+    def reconfig(self, model, config_list):
+        super().reconfig(model, config_list)
         self.now_epoch = 0
-        self.freq = epochs_per_iteration
-        self.end_epoch = epochs_per_iteration * num_iterations
-        self.set_wrappers_attribute("if_calculated", False)
```

Comment:
where is this line moved to?

Reply:
we have set this in `DependencyAwarePruner.__init__()`.